src/python/pants/backend/awslambda/python/rules.py

```diff
@@ -54,6 +55,14 @@ class PythonAwsLambdaFieldSet(PackageFieldSet):
 async def package_python_awslambda(
     field_set: PythonAwsLambdaFieldSet, lambdex: Lambdex, union_membership: UnionMembership
 ) -> BuiltPackage:
+
+    if sys.platform == "darwin":
```

Comment:
More idiomatic is to request `platform: Platform` (from `pants.engine.platform`) in the `@rule` args. Then you can check `platform.is_macos`

That way we use the engine to make the sys-call.